sklearnex/tests/test_run_to_run_stability.py

```diff
@@ -228,6 +259,14 @@ def test_other_estimator_stability(estimator, method, dataframe, queue):
         pytest.skip(f"variation observed in KMeans.score")
     if "NearestNeighbors" in estimator and "radius" in method:
         pytest.skip(f"RadiusNeighbors estimator not implemented in sklearnex")
+    if (
+        not _IS_INTEL
+        and ("Neighbors" in estimator or "LocalOutlierFactor" in estimator)
+        and method in ["score", "predict", "kneighbors", "kneighbors_graph"]
+    ):
+        if daal_check_version((2025, "P", 200)):
+            pytest.fail("Re-verify failure of algorithms in 2025.2 oneDAL")
+        pytest.skip(f"{estimator} shows instability on Non-Intel hardware")

```

Comment:
I propose to make a common function for checking and disabling as the nature of these crashes is the same. 
Also less code duplication

```python
def _skip_neighbors(estimator, estimator):
    if (
        not _IS_INTEL
        and ("Neighbors" in estimator or "LocalOutlierFactor" in estimator)
        and method in ["score", "predict", "kneighbors", "kneighbors_graph"]
    ):
        if daal_check_version((2025, "P", 200)):
            pytest.fail("Re-verify failure of algorithms in 2025.2 oneDAL")
        pytest.skip(f"{estimator} shows instability on Non-Intel hardware")
```

```suggestion
    _skip_neighbors(estimator, estimator)
```